BuildOpenSSL.cmake

```diff
@@ -0,0 +1,50 @@
+include(ExternalProject)
+set(OPENSSL_SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/_deps/openssl-src) # default path by CMake
+set(OPENSSL_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/openssl)
+set(OPENSSL_INCLUDE_DIR ${OPENSSL_INSTALL_DIR}/include)
+
+if (WIN32)
+    set(OPENSSL_CONFIGURE_COMMAND perl ${OPENSSL_SOURCE_DIR}/Configure VC-WIN64A)
+    set(OPENSSL_BUILD_COMMAND nmake /S)
+    set(OPENSSL_INSTALL_COMMAND nmake install /S)
+    set(OPENSSL_LIBRARY_EXTENSION .lib)
```

Comment:
@nathanklick these changes here involve the installed `perl` and `nmake` on our windows build machines. I had to update the %PATH% variable in order to access these executables. I'm not sure if this is the right way to do it so any feedback here would be appreciated.